terraform/eks/daemon/awsneuron/main.tf

```diff
@@ -0,0 +1,826 @@
+            },
+            {
+              type = "execution_stats"
+            }
+          ]
+        }
+      ]
+      system_metrics = [
+        {
+          type = "memory_info"
+        },
+        {
+          period = "5s"
+          type   = "neuron_hw_counters"
+        }
+      ]
+    })
+  }
+}
+
+resource "kubernetes_service_account" "neuron_monitor_service_account" {
+  depends_on = [
+    kubernetes_namespace.namespace
+  ]
+  metadata {
+    name      = "neuron-monitor-service-acct"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_role" "neuron_monitor_role" {
+  depends_on = [
+    kubernetes_namespace.namespace,
+    kubernetes_service_account.neuron_monitor_service_account,
+    kubernetes_config_map.neuron_monitor_config_map
+  ]
+  metadata {
+    name      = "neuron-monitor-role"
+    namespace = "amazon-cloudwatch"
+  }
+
+  rule {
+    api_groups     = [""]
+    resources      = ["configmaps"]
+    resource_names = ["neuron-monitor-config-map"]
+    verbs          = ["get"]
+  }
+}
+
+resource "kubernetes_role_binding" "neuron_monitor_role_binding" {
+  depends_on = [
+    kubernetes_namespace.namespace,
+    kubernetes_service_account.neuron_monitor_service_account,
+    kubernetes_role.neuron_monitor_role
+  ]
+
+  metadata {
+    namespace = "amazon-cloudwatch"
+    name      = "neuron-monitor-role-binding"
+  }
+
+  role_ref {
+    kind      = "Role"
+    name      = "neuron-monitor-role"
+    api_group = "rbac.authorization.k8s.io"
+  }
+
+  subject {
+    kind      = "ServiceAccount"
+    name      = "neuron-monitor-service-acct"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_daemonset" "neuron_monitor" {
+  depends_on = [
+    kubernetes_namespace.namespace,
+    kubernetes_service_account.neuron_monitor_service_account,
+    kubernetes_role.neuron_monitor_role,
+    kubernetes_role_binding.neuron_monitor_role_binding,
+    kubernetes_config_map.neuron_monitor_config_map
+  ]
+
+  metadata {
+    name      = "neuron-monitor"
+    namespace = "amazon-cloudwatch"
+    labels = {
+      k8s-app = "neuron-monitor"
+      version = "v1"
+    }
+  }
+  spec {
+    selector {
+      match_labels = {
+        k8s-app = "neuron-monitor"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          k8s-app = "neuron-monitor"
+          version = "v1"
+        }
+      }
+      spec {
+        affinity {
+          node_affinity {
+            required_during_scheduling_ignored_during_execution {
+              node_selector_term {
+                match_expressions {
+                  key      = "kubernetes.io/os"
+                  operator = "In"
+                  values   = ["linux"]
+                }
+              }
+            }
+          }
+        }
+        container {
+          name  = "neuron-monitor-prometheus"
+          image = "506463145083.dkr.ecr.us-west-2.amazonaws.com/mocked-neuron-monitor:v2"
```

Comment:
does this need to be updated everytime there is a new version of neuron monitor? what is the process of creating a mocked image?

Reply:
should be possible but we wanted to keep it as close as possible to the real implementation and deployment of the neuron monitor pod

Reply:
feels like it's another thing to manage outside of test itself when there is already a script doing exactly the same thing. should we expect this test to still pass when we use an actual image?

Reply:
it would, the only problem is it wont emit some of the metrics which can be sparse like errors and error correction events so we might miss some regression there. This way we inject those metrics in the prom script and then assert that all the metrics are correct

Reply:
> should be possible but we wanted to keep it as close as possible to the real implementation and deployment of the neuron monitor pod

How does it make close to the real implementation when the extra layer of docker image is just a wrapper of the python script underneath? I still think just running the same python script directly on a container or using the real neuron image would be better from the management perspective. With the real neuron image, is that sparse metric issue a known issue with no easy workaround or fix for it?

Reply:
Yeah, it was a conscious decision we took to not emit continuous 0s for metrics which  rarely occur, another case is that we want to simulate multiple runtimes running on the same host which we cannot test unless we actually run multiple runtimes (neuron burns) or mock their output.